pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -464,6 +499,102 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
 	return false, nil
 }
 
+// convertMAPIMachineOwnerReferencesToCAPI converts MAPI machine ownerReferences to CAPI ownerReferences.
+//
+//nolint:dupl
+func (r *MachineSyncReconciler) convertMAPIMachineOwnerReferencesToCAPI(ctx context.Context, mapiMachine *machinev1beta1.Machine) ([]metav1.OwnerReference, error) {
+	capiOwnerReferences := []metav1.OwnerReference{}
+
+	if len(mapiMachine.OwnerReferences) > 1 {
+		return nil, field.TooMany(field.NewPath("metadata", "ownerReferences"), len(mapiMachine.OwnerReferences), 1)
+	}
+
+	if len(mapiMachine.OwnerReferences) == 0 {
+		return capiOwnerReferences, nil
+	}
+
+	mapiOwnerReference := mapiMachine.OwnerReferences[0]
+	if mapiOwnerReference.Kind != machineSetKind || mapiOwnerReference.APIVersion != machinev1beta1.GroupVersion.String() {
+		return nil, field.Invalid(field.NewPath("metadata", "ownerReferences"), mapiMachine.OwnerReferences, errUnsuportedOwnerKindForConversion.Error())
+	}
+
+	key := types.NamespacedName{
+		Namespace: r.CAPINamespace,
+		Name:      mapiOwnerReference.Name,
+	}
+
+	capiMachineSet := capiv1beta1.MachineSet{}
+	// Get the CAPI machineSet with same name as MAPI machineSet
+	if err := r.Get(ctx, key, &capiMachineSet); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("could not find Cluster API machine set: %w", err)
+		} else {
+			return nil, fmt.Errorf("error getting Cluster API machine set: %w", err)
+		}
+	}
+
+	capiOwnerReference := metav1.OwnerReference{
+		Kind:               capiMachineSet.Kind,
+		APIVersion:         capiMachineSet.APIVersion,
+		Name:               capiMachineSet.Name,
+		Controller:         mapiOwnerReference.Controller,
+		BlockOwnerDeletion: mapiOwnerReference.BlockOwnerDeletion,
+		UID:                capiMachineSet.UID,
+	}
+
+	capiOwnerReferences = append(capiOwnerReferences, capiOwnerReference)
+
+	return capiOwnerReferences, nil
+}
+
+// convertCAPIMachineOwnerReferencesToMAPI converts CAPI machine ownerReferences to MAPI ownerReferences.
+//
+//nolint:dupl
+func (r *MachineSyncReconciler) convertCAPIMachineOwnerReferencesToMAPI(ctx context.Context, capiMachine *capiv1beta1.Machine) ([]metav1.OwnerReference, error) { //nolint:unused
```

Comment:
Is this used somewhere?

Reply:
This will be called from reconcileCAPIMachinetoMAPIMachine whe implmeneted.